scripts/test.sh

```diff
@@ -3,22 +3,22 @@ set -e
 
 # Brings in: ROOT_DIR, EXAMPLES_DIR, LIB_DIR, CODEGEN_DIR, CONTRIB_DIR, DOC_DIR
 SCRIPT_DIR="$( cd "$( dirname "${BASH_SOURCE[0]}" )" && pwd )"
-source $SCRIPT_DIR/config.sh
+source "${SCRIPT_DIR}/config.sh"
 
 # Add Cargo to PATH.
 export PATH=${HOME}/.cargo/bin:${PATH}
 
 # Checks that the versions for Cargo projects $@ all match
 function check_versions_match() {
   local last_version=""
-  for dir in $@; do
+  for dir in "$@"; do
```

Comment:
Use `${@}`.

Actually, do we even need to use quotes here? The inputs should be properly separated already. Or will they be re-separated here?